iot/api/CloudIotMqttExample/Program.cs

```diff
@@ -0,0 +1,645 @@
+            client.MqttMsgPublishReceived += Client_MqttMsgPublishReceived;
+            client.MqttMsgPublished += Client_MqttMsgPublished;
+
+            return client;
+        }
+        // [END iot_mqtt_client]
+
+        public static object SetupMqttTopics(MqttClient client, string deviceId)
+        {
+            // This is the topic that the device will receive configuration updates on.
+            string mqttConfigTopic = $"/devices/{deviceId}/config";
+            // The topic that the device will receive commands on.
+            string mqttCommandTopic = $"/devices/{deviceId}/commands/#";
+            string mqttErrorTopic = $"/devices/{deviceId}/errors";
+
+            string[] topics = new string[] {
+                mqttConfigTopic,
+                mqttCommandTopic,
+                mqttErrorTopic
+            };
+
+            byte[] qosLevels = new byte[] {
+                MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE , // config topic, QoS 1 enables message acknowledgement.
+                MqttMsgBase.QOS_LEVEL_AT_MOST_ONCE , // command topic
+                MqttMsgBase.QOS_LEVEL_AT_MOST_ONCE   // error topic
+            };
+            // Subscribe to the config topic, and command topic.
+            Console.WriteLine("Subscribing to {0}", mqttCommandTopic);
+            client.Subscribe(topics, qosLevels);
+            return 0;
+        }
+
+        // [START iot_mqtt_event_handlers]
+        public static void Client_ConnectionClosed(object sender, EventArgs e)
+        {
+            Console.WriteLine("On Disconnect: {0} {1}", e.GetType(), sender);
+        }
+
+        public static void Client_MqttMsgSubscribed(object sender, MqttMsgSubscribedEventArgs e)
+        {
+            Console.WriteLine("On Subscribe: {0} {1}", e.MessageId, sender);
+        }
+
+        public static void Client_MqttMsgPublished(object sender, MqttMsgPublishedEventArgs e)
+        {
+            Console.WriteLine("On Publish {0}", e.IsPublished);
+        }
+        // the event MqttMsgPublishReceived raised when a message is published on a topic the client is subscribed to.
+        public static void Client_MqttMsgPublishReceived(object sender, MqttMsgPublishEventArgs e)
+        {
+            // handle message received 
+            var output = $"Received { System.Text.Encoding.UTF8.GetString(e.Message)} on topic {e.Topic} with Qos {e.QosLevel}";
+            Console.WriteLine(output);
+        }
+        // [END iot_mqtt_event_handlers]
+
+        // [START iot_mqtt_example]
+        public static object StartMqtt(string projectId, string cloudRegion, string registryId, string deviceId,
+         string privateKeyFile, string algorithm, string caCert, int numMsgs, string messageType, string mqttBridgeHostName,
+            int mqttBridgePort, int jwtExpiresMin, int waitTime)
+        {
+            //setting default values if not passed from parameters
+            numMsgs = numMsgs <= 0 ? 20 : numMsgs;
```

Comment:
C# has default values for function arguments.  In other words, 

```
        public static object StartMqtt(string projectId, string cloudRegion, string registryId, string deviceId,
         string privateKeyFile, string algorithm = "sha256", string caCert = "approved", int numMsgs = 20, string messageType = "postcard", string mqttBridgeHostName = "JeffsHost.com",
            int mqttBridgePort = 10, int jwtExpiresMin = 10, int waitTime = 10)
```
I think that would be more readable?

Reply:
Thats super neat, I will definitely replace with that.